src/components/DropdownSelect2/DropdownSelect2.js

```diff
@@ -0,0 +1,329 @@
+import React, {
+  useCallback,
+  Fragment,
+  useRef,
+  useState,
+  useEffect
+} from 'react'
+import PropTypes from 'prop-types'
+import Error from '@components/Form/Error'
+import List from '@components/List'
+import ListItem from '@components/List/Item'
+import Dropdown from '@components/Dropdown'
+import Heading from '@components/Heading'
+import Button from '@components/Button'
+import Overflow from '@components/Overflow'
+import Divider from '@components/Divider'
+import Loader from '@components/Loader'
+import { emitCloseDropdown } from '@events/dropdown'
+import { isEmpty } from '@helpers/data'
+import Spacer from '@components/Spacer'
+import Searcher from '@components/Searcher'
+import Paragraph from '@components/Paragraph'
+import Label from '@components/Label'
+import styles from './DropdownSelect.module.scss'
+import { useStyles } from '@helpers/hooks/useStyles'
+
+/**
+ * DropdownSelect - stateless presentational component
+ * @param {object} props - props
+ * @param {func} props.className - wrapper custom styles
+ * @param {string} props.value - field value
+ * @param {func} props.onChange - on Change handler
+ * @param {object} props.errors - element errors list
+ * @param {object} props.touched - element touched list
+ * @param {string} props.label - label
+ * @param {array} props.options - list of options
+ * @param {array} props.emphasisedOptions - list of emphasised options, displayed on top
+ * @param {array} props.hasDescription - render description in options
+ * @param {string} props.hasSearcher - show Searcher
+ * @param {func} props.handleOnSearchChange - on search input change handler
+ * @param {bool} props.isLoading - list loading spinner
+ * @param {object} props.emptyMessage - empty message component
+ * @param {bool} props.isOpenDisabled - when its true dropdown can't be open, default: false
+ * @param {bool} props.optionalContent - optional content to render on bottom
+ * @param {bool} props.alwaysShowLabel - always show label on top
+ * @param {object} props.overflowStyle - overflow styles
+ * @param {func} props.formikKey - name on formik 'nasted' keys
+ * @param {string} props.i18n - object of translations
+ * @param {string} props.inModalName - modal name for positioning
+ * @return {object} An object of children element
+ */
+const DropdownSelect = ({
+  className,
+  value,
+  onChange,
+  errors,
+  touched,
+  label,
+  options,
+  emphasisedOptions,
+  hasDescription,
+  hasSearcher,
+  handleOnSearchChange,
+  isLoading,
+  emptyMessage,
+  isOpenDisabled,
+  optionalContent,
+  alwaysShowLabel,
+  overflowStyle,
+  formikKey,
+  i18n,
+  inModalName
+}) => {
+  const hasLabel = value || alwaysShowLabel
+
+  const labelStyles = useStyles({
+    [styles['form-field__label']]: true,
+    [styles['form-field__label--has-error']]:
+      errors[formikKey] || touched[formikKey],
```

Comment:
ogólnie jest okej, ale jak ten key formika będzie zagnieżdzony to to nie zadziała. Oskar kiedyś rozwiązywał ten problem ale nie pamiętam gdzie ://